activerecord/test/cases/inheritance_test.rb

```diff
@@ -352,7 +352,7 @@ def test_inheritance_without_mapping
   end
 
   def test_scope_inherited_properly
-    assert_nothing_raised { Company.of_first_firm }
+    assert_nothing_raised { Company.of_first_firm.to_sql }
     assert_nothing_raised { Client.of_first_firm }
```

Comment:
Triggers the `JoinDependency` to reflect on the associations and raise when `account` can't be found on `Company`.

Should also apply to line 356.